src/core/lib/security/security_connector/tls/tls_security_connector.h

```diff
@@ -145,10 +145,32 @@ class TlsServerSecurityConnector final : public grpc_server_security_connector {
 };
 
 // ---- Functions below are exposed for testing only -----------------------
+
+/** The |TlsFetchKeyMaterials| API ensures that |key_materials_config| has a
+ *  non-empty pem-key-cert pair list. This is done as follows:
+ *  - if |options| is equipped with a credential reload config, then this
+ *    methods uses credential reloading to populate |key_materials_config|, and
+ *    afterwards it populates |reload_status| with the status of this operation.
+ *    In particular, any data stored in |key_materials_config| is overwritten.
+ *  - if |options| has no credential reload config, then:
+ *    - if |key_materials_config| already has a non-empty pem-key-cert pair
+ *      list or is called by a client, then the method returns |GRPC_STATUS_OK|.
+ *    - if |key_materials_config| has an empty pem-key-cert pair list and is
+ *      called by a server, then the method return an error code.
+ *
+ *  The arguments are detailed below:
+ *  - key_materials_config: a key material config that will be populated by the
```

Comment:
Probably it is also worth documenting "`key_materials_config` is a reference to the credentials currently being used"?

Reply:
Sorry I don't follow what you mean. Would you like me to explain the kind of data that the `key_materials_config` holds?

Reply:
Yes. I was suggesting that because when looking at the comments, I was kind of confused about where `key_materials_config` is passed from, and soon realized it is actually a pointer to the same `key_materials_config` in security_connector. I thought it might also be worth documenting where this variable was from(or in other way, what is the current value before we call `TlsFetchKeyMaterials`), besides how it was going to be used. 
This is just a minor comment and I am fine with keeping it, too :-)

Reply:
I think it is better for the description of the API to only assert what it does, and not how it is used by the other methods. Is there somewhere else in the code where such a comment could be added that would have made it more clear? (For example, in the implementation of e.g. `TlsChannelSecurityConnector::InitializeHandshakerFactory`?)

Reply:
Yes, I agree. Probably it is better to add these comments in the caller like `InitializeHandshakerFactory`.

Reply:
Sounds good. I added comments to each of the 4 calls to `TlsFetchKeyMaterials`. PTAL.

Reply:
LGTM. Thanks Matt!